runsc/boot/compat.go

```diff
@@ -86,36 +84,8 @@ func (c *compatEmitter) Emit(msg proto.Message) (bool, error) {
 }
 
 func (c *compatEmitter) emitUnimplementedSyscall(us *spb.UnimplementedSyscall) {
-	regs := us.Registers.GetArch().(*rpb.Registers_Amd64).Amd64
-
-	c.mu.Lock()
-	defer c.mu.Unlock()
-
-	sysnr := regs.OrigRax
```

Comment:
You could keep this method here and create a function to return the syscall number from regs in the arch specific files.

Reply:
The "syscall number" is not the only difference here, and the syscall.SYS_ARCH_PRCTL is not supported on arm64.

Reply:
Yes, making a call to an arch specific function if there are no matches here would be preferable to reduce duplication as much as possible.

Reply:
Done, thanks!